app/components/avo/modal_component.rb

```diff
@@ -1,22 +1,19 @@
 # frozen_string_literal: true
 
-class Avo::ModalComponent < ViewComponent::Base
-  renders_one :heading
-  renders_one :controls
+class Avo::ModalComponent < Avo::BaseComponent
+  SIZE = _Union(:md, :xl)
 
-  attr_reader :width
-  attr_reader :body_class
+  prop :width, SIZE, default: :md
+  prop :body_class, _Nilable(String)
 
-  def initialize(width: :md, body_class: nil)
-    @width = width
-    @body_class = body_class
-  end
+  renders_one :heading
+  renders_one :controls
 
   def width_classes
-    case width.to_sym
-    when :md
+    case @width
+    in :md
       "w-11/12 lg:w-1/2 sm:max-w-168"
-    when :xl
+    in :xl
```

Comment:
I switched this to case-in, since it’s more exhaustive. It has a default else-raise.